src/app/sidebar-tree/sidebar-tree.component.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {MatTreeNestedDataSource} from '@angular/material/tree';
+import {NestedTreeControl} from '@angular/cdk/tree';
+import {Category} from '../category';
+import {of} from 'rxjs';
+import {ActivatedRoute} from '@angular/router';
+
+@Component({
+  selector: 'app-sidebar-tree',
+  templateUrl: './sidebar-tree.component.html',
+  styleUrls: ['./sidebar-tree.component.css']
+})
+export class SidebarTreeComponent implements OnInit, OnDestroy {
+
+  nestedTreeControl: NestedTreeControl<Category>;
+  nestedDataSource: MatTreeNestedDataSource<Category>;
+  categories: [Category];
+  navigationSubscription: any;
+
+  constructor(private activatedRoute: ActivatedRoute) {
+    this.nestedTreeControl = new NestedTreeControl<Category>(this._getChildren);
+    this.nestedDataSource = new MatTreeNestedDataSource();
+  }
+
+  initialiseInvites() {
+    this.nestedDataSource.data = this.activatedRoute.snapshot.data['categories'];
+  }
+
+  ngOnInit() {
+    this.initialiseInvites();
+  }
+
+  ngOnDestroy() {
+    if (this.navigationSubscription) {
+      this.navigationSubscription.unsubscribe();
+    }
+  }
```

Comment:
It seems to me that `navigationSubscription` is redundant and unused in this component